ptf/tests/srv6.py

```diff
@@ -358,71 +345,46 @@ def testPacket(self, pkt, sid_list):
 
         # Build expected packet from the given one...
         exp_pkt = pkt.copy()
-        # Route MAC addresses
+
+        # Set IPv6 dst to next SID and decrement segleft...
+        exp_pkt[IPv6].dst = next_sid
+        exp_pkt[IPv6ExtHdrSegmentRouting].segleft -= 1
+
+        # Route and decrement TTL...
         pkt_route(exp_pkt, next_hop_mac)
-        # Set IPv6 dst to first SID...
-        exp_pkt[IPv6].dst = sid_list[0]
-        # Decrement TTL...
         pkt_decrement_ttl(exp_pkt)
-        # Insert SRv6 header between IPv6 header and payload
-        exp_pkt[IPv6ExtHdrSegmentRouting].segleft = exp_pkt[IPv6ExtHdrSegmentRouting].segleft - 1
-
-        # FIXME: the P4 pipeline should calculate correct checksum
-        if TCP in pkt:
-            pkt[TCP].chksum = 1
-            exp_pkt[TCP].chksum = 1
-        if UDP in pkt:
-            pkt[UDP].chksum = 1
-            exp_pkt[UDP].chksum = 1
-        if ICMPv6Unknown in exp_pkt:
-            pkt[ICMPv6Unknown].cksum = 1
-            exp_pkt[ICMPv6Unknown].cksum = 1
+
+        # FIXME: update P4 program to calculate correct checksum
+        set_cksum(pkt, 1)
+        set_cksum(exp_pkt, 1)
 
         testutils.send_packet(self, self.port1, str(pkt))
         testutils.verify_packet(self, exp_pkt, self.port2)
 
 
 @group("srv6")
-class FabricSrv6EndPspTest(P4RuntimeTest):
-    """Tests SRv6 end PSP behavior"""
-
-    def buildSrv6Pkt(self, pkt, sid_list):
-        sid_len = len(sid_list)
-        # Build srv6 packet from the given one...
-        srv6_pkt = pkt.copy()
-        # Route MAC addresses
-        pkt_route(srv6_pkt, SWITCH3_MAC)
-        # Set IPv6 dst to last SID...
-        srv6_pkt[IPv6].dst = SWITCH3_IPV6
-        # Insert SRv6 header between IPv6 header and payload
-        srv6_hdr = IPv6ExtHdrSegmentRouting(
-            nh=pkt[IPv6].nh,
-            addresses=sid_list[::-1],
-            len=sid_len * 2,
-            segleft=1,
-            lastentry=sid_len - 1)
-        srv6_pkt[IPv6].nh = 43  # next header is SR header
-        srv6_pkt[IPv6].payload = srv6_hdr / pkt[IPv6].payload
-        return srv6_pkt
+class Srv6EndPspTest(P4RuntimeTest):
+    """Tests SRv6 end PSP behavior (i.e. with pop), where the switch SID is the
+    last one and the switch removes the SRv6  header beofre routing the packet
+    to  it's final destination.
```

Comment:
`where the switch SID is the last one`
The host IP is the last one.
I would use something like `where segleft is equals to 1`